node/forest_libp2p/src/blocksync/provider.rs

```diff
@@ -0,0 +1,241 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use chain::Error as ChainError;
+use forest_blocks::{Tipset, TipsetKeys};
+use forest_cid::Cid;
+use ipld_blockstore::BlockStore;
+use log::debug;
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use super::{
+    BlockSyncRequest, BlockSyncResponse, BlockSyncResponseStatus, CompactedMessages, TipsetBundle,
+};
+
+/// Builds blocksync response out of chain data.
+pub fn make_blocksync_response<DB>(db: Arc<DB>, request: &BlockSyncRequest) -> BlockSyncResponse
```

Comment:
This is a small detail, but you don't need an `Arc` to be passed in here, only need a reference. I assumed you did because it's done in a new task, and you would need to clone to move the `Arc` into that task, but you don't need to move in the `Arc` into this function. Only reason I mention this is that it makes it easier to test if you just need to pass in a db reference

Reply:
I will double-check on this tonight, but I can recall that there was an error along the lines of `Sync+Send` is not implemented for `&DB` when attempting to pass a reference only.

Reply:
Yeah so you have to still clone the arc, but you only need to pass a reference to the cloned `Arc<DB>` to the function

Reply:
yes, just tested and it works :+1: Will update today.